### PR TITLE
More reduce undisposed CancellationTokenSource

### PR DESCRIPTION
**Pull request by longfin**
This PR adds handling for `CancellationTokenSource.Dispose()` to prevent memory leakages.

### CODE REVIEW

**Review by limebell — APPROVED** [review]
👍

**Review by moreal — APPROVED** [review]

File: Libplanet/Blockchain/BlockChain.cs [hunk 1]
```diff
@@ -960,8 +960,6 @@ public long GetNextTxNonce(Address address)
             finally
             {
                 TipChanged -= WatchTip;
-                cancellationTokenSource.Dispose();
-                cts.Dispose();
```
Comment (riemannulus) [inline Libplanet/Blockchain/BlockChain.cs:964]: It wasn't disposed of here?

Comment (longfin) [reply Libplanet/Blockchain/BlockChain.cs:964]: it can be bypassed if exceptions were raised before `try` block.

**Review by riemannulus — APPROVED** [review]

File: Libplanet/Net/Swarm.cs [hunk 2]
```diff
@@ -901,10 +910,11 @@ await foreach (var pair in demandBlockHashes)
...
             CancellationToken cancellationToken = default(CancellationToken))
         {
-            cancellationToken = CancellationTokenSource
-                .CreateLinkedTokenSource(cancellationToken, _cancellationToken).Token;
+            using CancellationTokenSource cts = CancellationTokenSource
+                .CreateLinkedTokenSource(cancellationToken, _cancellationToken);
+            cancellationToken = cts.Token;
 
-            var kademliaProtocol = (KademliaProtocol)PeerDiscovery;
+            KademliaProtocol kademliaProtocol = (KademliaProtocol)PeerDiscovery;
```
Comment (dahlia) [inline Libplanet/Net/Swarm.cs:917]: It's trivial but seems redundant as the casting operator indicates the type.

**Review by dahlia — APPROVED** [review]

**Pull request merged**